jax/_src/numpy/polynomial.py

```diff
@@ -175,21 +175,79 @@ def polyfit(x: Array, y: Array, deg: int, rcond: float | None = None,
     return c
 
 
-_POLY_DOC = """\
-This differs from np.poly when an integer array is given.
-np.poly returns a result with dtype float64 in this case.
-jax returns a result with an inexact type, but not necessarily
-float64.
-
-This also differs from np.poly when the input array strictly
-contains pairs of complex conjugates, e.g. [1j, -1j, 1-1j, 1+1j].
-np.poly returns an array with a real dtype in such cases.
-jax returns an array with a complex dtype in such cases.
-"""
-
-@implements(np.poly, lax_description=_POLY_DOC)
 @jit
 def poly(seq_of_zeros: Array) -> Array:
+  r"""Returns the coefficients of a polynomial for the given sequence of roots.
+
+  JAX implementation of :func:`numpy.poly`.
+
+  Args:
+    seq_of_zeros: A scalar or an array of roots of the polynomial of shape ``(M,)``
+      or ``(M, M)``.
+
+  Returns:
+    An array containing the coefficients of the polynomial.
+
+  Note:
+
+    :func:`jax.numpy.poly` differs from :func:`numpy.poly` in the following scenarios:
+
+    - When the input is a scalar, ``np.poly`` raises a ``TypeError``, whereas
+      ``jnp.poly`` returns a result with an inexact type, but not necessarily
+      float64.
+    - When an integer array is given, ``np.poly`` returns a result with dtype
+      float64, whereas ``jnp.poly`` returns a result with an inexact type, but
+      not necessarily float64.
+    - When the input array strictly contains pairs of complex conjugates, e.g.
+      [1j, -1j, 1-1j, 1+1j], ``np.poly`` returns an array with a real dtype
+      whereas ``jnp.poly`` returns an array with a complex dtype.
+    - When the input array is a real valued 2D square matrix, ``np.poly`` 
+      returns an array with a real dtype whereas ``jnp.poly`` returns an
+      array with a complex dtype.
+
+  See also:
+    - :func:`jax.numpy.polyfit`: Least squares polynomial fit.
+    - :func:`jax.numpy.polyval`: Evaluate a polynomial at specific values.
+    - :func:`jax.numpy.roots`: Computes the roots of a polynomial for given
+      coefficients.
+
+  Example:
+
+    Scalar inputs:
+
+    >>> np.poly(1)  # doctest: +IGNORE_EXCEPTION_DETAIL
+    Traceback (most recent call last):
+      File "<stdin>", line 1, in <module>
+    TypeError: dispatcher for __array_function__ did not return an iterable
+    >>> jnp.poly(1)
+    Array([ 1., -1.], dtype=float32)
+
+    Input array with integer values:
+
+    >>> x = jnp.array([1, 2, 3])
+    >>> np.poly(x)
```

Comment:
No reason to show numpy output in JAX examples. Same below.

Use the examples to show how the function is used, and why it might be relevant to the user, not to show nitpicky details about how JAX's implementation differs from NumPy's

Same below.

Reply:
Removed numpy examples